buffer.c

```diff
@@ -0,0 +1,278 @@
+/**********************************************************************
+
+  buffer.c - Buffer Protocol
+
+  Copyright (C) 2020 Kenta Murata <mrkn@mrkn.jp>
+
+**********************************************************************/
+
+#include "ruby/internal/config.h"
+
+static ID id_buffer_protocol;
+
+static const rb_data_type_t random_mt_type = {
+    "buffer_protocol",
+    {
+	0,
+	0,
+	0,
+    },
+    0, 0, RUBY_TYPED_FREE_IMMEDIATELY
+};
+
+/* Register buffer protocol functions for the given class */
+int
+rb_buffer_protocol_register_klass(VALUE klass, const rb_buffer_protocol_entry_t *entry) {
```

Comment:
How about removing `_klass` suffix?
We don't have any other variants of this feature.